mitxgraders/sampling.py

```diff
@@ -210,12 +222,12 @@ def gen_sample(self):
         return self.modulus.gen_sample() * np.exp(1j * self.argument.gen_sample())
 
 
-class DiscreteSet(VariableSamplingSet):  # pylint: disable=too-few-public-methods
+class DiscreteSet(ScalarSamplingSet):  # pylint: disable=too-few-public-methods
```

Comment:
Prior to this PR, `DiscreteSet` could sample from a list of matrices, right? 

I understand that this change lets `DiscreteSet` be used as a sampler for `IdentityMatrixMultiples`. But if you need this feature, it seems better to me to keep DiscreteSet general (since its name sounds general) and have a specialized DiscreteScalarSet. 

Or, if you don't need this feature, just remove DiscreteSet as one of the possible IdentityMatrixMultiples samplers, and we can cross this bridge when necessary.

Reply:
It actually can't. `DiscreteSet` is defined using the schema `schema_config = Schema(TupleOfType(Number))`, so if we want to be able to have matrices here, we need to update `DiscreteSet`. However, I think your point is valid: having matrices in `DiscreteSet` obviates the need to be able to use `DiscreteSet` in `IdentityMatrixSampler` (just pass in the appropriate matrices to `DiscreteSet`!)